elasticgraph-warehouse/lib/elastic_graph/warehouse/schema_definition/schema_artifact_manager_extension.rb

```diff
@@ -0,0 +1,49 @@
+# Copyright 2024 - 2025 Block, Inc.
+#
+# Use of this source code is governed by an MIT-style
+# license that can be found in the LICENSE file or at
+# https://opensource.org/licenses/MIT.
+#
+# frozen_string_literal: true
+
+require "elastic_graph/warehouse"
+
+module ElasticGraph
+  module Warehouse
+    module SchemaDefinition
+      # Extension module for {ElasticGraph::SchemaDefinition::SchemaArtifactManager} that adds
+      # warehouse artifact generation support.
+      #
+      # @private
+      module SchemaArtifactManagerExtension
+        private
+
+        # Overrides the base `artifacts_from_schema_def` method to add warehouse artifacts.
+        #
+        # This method is called when computing the list of schema artifacts. It calls super
+        # to get the base artifacts, then appends the warehouse artifact if any warehouse
+        # tables are defined.
+        #
+        # @return [Array<ElasticGraph::SchemaDefinition::SchemaArtifact>] the list of schema artifacts
+        def artifacts_from_schema_def
+          base_artifacts = super
+          # `__skip__` is needed because `schema_definition_results` returns `Results`, but at runtime
+          # it will have been extended with `ResultsExtension` which provides `warehouse_config`.
+          # @type var warehouse_config: ::Hash[::String, ::Hash[::String, untyped]]
+          __skip__ = warehouse_config = schema_definition_results.warehouse_config
```

Comment:
```suggestion
          results = schema_definition_results # : ElasticGraph::SchemaDefinition::Results & ResultsExtension
          warehouse_config = results.warehouse_config
```

Here's a way to get it to type check w/o needing to `__skip__`:

* Introduce a new variable.
* Tell Steep what it's type is.